driver-core/src/main/java/com/datastax/driver/core/AbstractAddressableByIndexData.java

```diff
@@ -256,22 +209,14 @@ public T setInet(int i, InetAddress v) {
     }
 
     public T setUDTValue(int i, UDTValue v) {
-        checkType(i, UDT);
         return setValue(i, codecFor(i, UDTValue.class).serialize(v, protocolVersion));
     }
 
     public T setTupleValue(int i, TupleValue v) {
-        checkType(i, TUPLE);
         return setValue(i, codecFor(i, TupleValue.class).serialize(v, protocolVersion));
     }
 
     @Override
-    public <V> T setObject(int i, V v) {
```

Comment:
Thanks for the explanation.  Using `CodecRegistry#codecFor(DataType, T)` sounds good, in this case the user needs to know the DataType, but they should probably know this anyways, and if not they could use `TableMetadata` to get at that.